fv3core/_config.py

```diff
@@ -204,7 +204,7 @@ def use_logp(self) -> bool:
 
 
 @dataclasses.dataclass
-class Namelist:
+class DynamicalCoreConfig:
```

Comment:
A Namelist is a a common model concept, I think this might be an unnecessarily confusing rename for items that come directly from the namelist

Reply:
but it's fine, we've gone back and forth on this, we can make it module specific

Reply:
This is an error on my part, I should have a Namelist class (which is initialized only from a fortran namelist, also includes physics settings, and uses Fortran naming) and a separate DynamicalCoreConfig class (which can be initialized in multiple ways, contains only dycore namelist variables, and whose attributes we can rename). I'll add a Namelist class which at this point is mostly identical to DynamicalCoreConfig.